examples/lessons/11/app.js

```diff
@@ -0,0 +1,206 @@
+
+varying vec2 vTextureCoord;
+varying vec3 vLightWeighting;
+
+uniform sampler2D uSampler;
+
+void main(void) {
+  vec4 textureColor = texture2D(uSampler, vec2(1.0 - vTextureCoord.s, 1.0 - vTextureCoord.t));
+  gl_FragColor = vec4(textureColor.rgb * vLightWeighting, textureColor.a);
+}
+`;
+
+const mouseMovement = {
+  mouseDown: false,
+  lastMouseX: null,
+  lastMouseY: null
+};
+
+const moonRotation = {
+  matrix: new Matrix4()
+};
+
+const animationLoop = new AnimationLoop({
+  onInitialize: ({canvas, gl}) => {
+    addMouseHandler(canvas, mouseMovement, moonRotation);
+
+    setParameters(gl, {
+      clearColor: [0, 0, 0, 1],
+      clearDepth: 1,
+      depthTest: true
+    });
+
+    return loadTextures(gl, {
+      urls: ['moon.gif']
+    })
+    .then(textures => {
+      let program = new Program(gl, {
+        fs: FRAGMENT_SHADER,
+        vs: VERTEX_SHADER
+      });
+
+      let moon = new Sphere(gl, {
+        program,
+        uniforms: {
+          uSampler: textures[0]
+        },
+        nlat: 30,
+        nlong: 30,
+        radius: 2,
+      });
+      return {moon};
+    });
+  },
+  onRender: ({
+    gl, tick, aspect, moon
+  }) => {
+    // Update Camera Position
+    const eyePos = [0, 0, 6];
+
+    gl.clear(GL.COLOR_BUFFER_BIT | GL.DEPTH_BUFFER_BIT);
+
+    let uMMatrix = new Matrix4()
+      .multiplyRight(moonRotation.matrix);
+    let uVMatrix = new Matrix4()
+      .lookAt({eye: eyePos, center: [0, 0, 0], up:[0, 1, 0]});
+
+    let lighting = document.getElementById("lighting").checked;
+
+    moon.setUniforms({uUseLighting: lighting});
+
+    if (lighting) {
+      let ambientColor = new Vector3(
+        parseFloat(document.getElementById("ambientR").value),
+        parseFloat(document.getElementById("ambientG").value),
+        parseFloat(document.getElementById("ambientB").value)
+      );
+
+      let lightingDirection = new Vector3(
+        parseFloat(document.getElementById("lightDirectionX").value),
+        parseFloat(document.getElementById("lightDirectionY").value),
+        parseFloat(document.getElementById("lightDirectionZ").value)
+      );
+      lightingDirection.normalize();
+      lightingDirection.scale(-1);
+
+      let directionalColor = new Vector3(
+        parseFloat(document.getElementById("directionalR").value),
+        parseFloat(document.getElementById("directionalG").value),
+        parseFloat(document.getElementById("directionalB").value)
+      );
+
+      moon.setUniforms({
+        uAmbientColor: ambientColor,
+        uLightingDirection: lightingDirection,
+        uDirectionalColor: directionalColor
+      });
+    }
+
+    moon.render({
+      uMMatrix,
+      uVMatrix,
+      uPMatrix: new Matrix4().perspective({fov: 45 * Math.PI / 180, aspect, near: 0.1, far: 100})
+    });
+  }
+});
+
+animationLoop.getInfo = () => {
+  return `
+  <p>
+    <a href="http://learningwebgl.com/blog/?p=1253" target="_blank">
+      Improving the code structure with lots of moving objects
```

Comment:
Same comment as last PR: please update the title of the lesson.

In this case if you follow the link: "Spheres, Rotation Matrices, and Mouse Events" (instead of "Improving the code structure with lots of moving objects", which is Lesson 9 title)

Reply:
thanks, fixed both examples.